### PR TITLE
Alternative flight frontend updates

### PR DESCRIPTION
**Pull request by nowshin-108**
Frontend - Alternative Flight Recommendation Feature

This pull request includes: 
   - Created new components: `AltFlightModal.jsx` and `AltFlightModal.css` for displaying alternative flight options
   - Updated `TripDetails.jsx` and `Upcoming.jsx` to integrate with the new recommendation feature
Screenshot 2024-07-16 at 7 54 38 PM

### CODE REVIEW

**Comment by alantulais** [comment]
It seems to be this PR is including changes form previous backend PR?

File: frontend/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "@react-google-maps/api": "^2.19.3",
     "amadeus": "^10.1.0",
     "axios": "^1.7.2",
+    "date-fns": "^3.6.0",
```
Comment (alantulais) [inline frontend/package.json:17]: What's the usage of this API?

File: frontend/src/Components/HomePage/UpcomingTrips/Upcoming.jsx [hunk 2]
```diff
@@ -25,6 +30,21 @@ function Upcoming() {
         fetchTrips();
     }, []);
 
+    const moveToCompleted = async (trip) => {
```
Comment (alantulais) [inline frontend/src/Components/HomePage/UpcomingTrips/Upcoming.jsx:33]: Can we include this change in separate PR? Fetching recommended flights LGTM, but I'd like us to chat about this part more carefully. I don't think updating flight status on frontend behind the scenes is the way to go.

**Review by alantulais — CHANGES_REQUESTED** [review]
Overall LGTM. We can polish the UX later on, would recommend to add some pagination as well.

Back to you to remove the changes to update trip status, since I'd like us to re-think that approach.

File: frontend/package.json (hunk 1 above)
Comment (nowshin-108) [reply frontend/package.json:17]: It's used to format date

File: frontend/src/Components/HomePage/UpcomingTrips/Upcoming.jsx (hunk 2 above)
Comment (nowshin-108) [reply frontend/src/Components/HomePage/UpcomingTrips/Upcoming.jsx:33]: On it!

**Review by alantulais — APPROVED** [review]

**Pull request merged**